vortexor/tests/vortexor.rs

```diff
@@ -71,3 +84,92 @@ async fn test_vortexor() {
+}
+
+#[tokio::test(flavor = "multi_thread")]
+async fn test_stake_update() {
+    solana_logger::setup();
+
+    // Create a local cluster with 3 validators
+    let default_node_stake = 10 * LAMPORTS_PER_SOL; // Define a default value for node stake
+    let mint_lamports = 100 * LAMPORTS_PER_SOL;
+    let mut config = ClusterConfig::new_with_equal_stakes(3, mint_lamports, default_node_stake);
+
+    let mut cluster = LocalCluster::new(&mut config, SocketAddrSpace::Unspecified);
+    info!(
+        "Cluster created with {} validators",
+        cluster.validators.len()
+    );
+    assert_eq!(cluster.validators.len(), 3);
+
+    let pubkey = cluster.entry_point_info.pubkey();
+    let validator = &cluster.validators[pubkey];
+
+    let mut servers = vec![get_server_urls(validator)];
+    // add one more RPC subscription to another validator
+    for validator in cluster.validators.values() {
+        if validator.info.keypair.pubkey() != *pubkey {
+            servers.push(get_server_urls(validator));
+            break;
+        }
+    }
+    let exit = Arc::new(AtomicBool::new(false));
+
+    let (rpc_load_balancer, slot_receiver) =
+        rpc_load_balancer::RpcLoadBalancer::new(&servers, &exit);
+
+    // receive 2 slot updates
+    let mut i = 0;
+    let slot_receive_timeout = Duration::from_secs(5); // conservative timeout to ensure stable test
+    while i < 2 {
+        let slot = slot_receiver
+            .recv_timeout(slot_receive_timeout)
+            .unwrap_or_else(|_| panic!("Expected a slot within {slot_receive_timeout:?}"));
+        i += 1;
+        info!("Received a slot update: {}", slot);
+    }
+
+    let rpc_load_balancer = Arc::new(rpc_load_balancer);
+    // Now create a stake updater service
+    let shared_staked_nodes = Arc::new(RwLock::new(StakedNodes::default()));
+    let staked_nodes_updater_service = StakeUpdater::new(
+        exit.clone(),
+        rpc_load_balancer.clone(),
+        shared_staked_nodes.clone(),
+        Duration::from_millis(100), // short sleep to speed up the test for service exit
+    );
+
+    // Waiting for the stake map to be populated by the stake updater service
+    let start_of_stake_updater = std::time::Instant::now();
+    let stake_updater_timeout = Duration::from_secs(10); // conservative timeout to ensure stable test
+    loop {
+        let stakes = shared_staked_nodes.read().unwrap();
+        if let Some(stake) = stakes.get_node_stake(pubkey) {
+            info!("Stake for {}: {}", pubkey, stake);
+            assert_eq!(stake, default_node_stake);
+            let total_stake = stakes.total_stake();
+            info!("total_stake: {}", total_stake);
+            assert!(total_stake >= default_node_stake);
+            break;
+        }
+        info!("Waiting for stake map to be populated for {pubkey:?}...");
+        drop(stakes); // Drop the read lock before sleeping so the writer side can proceed
+        std::thread::sleep(std::time::Duration::from_millis(100));
```

Comment:
you should not call thread::sleep in async context, this can freeze the executor

Reply:
will fix